climada/engine/uncertainty_quantification/calc_impact.py

```diff
@@ -0,0 +1,268 @@
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Define Uncertainty Impact class
+"""
+
+__all__ = ['CalcImpact']
+
+import logging
+import time
+import copy
+
+import pandas as pd
+import numpy as np
+
+from climada.engine import Impact
+from climada.engine.uncertainty_quantification import Calc, InputVar
+from climada.util import log_level
+
+from climada.util.config import setup_logging as u_setup_logging
+
+LOGGER = logging.getLogger(__name__)
+u_setup_logging()
+
+
+class CalcImpact(Calc):
+    """
+    Impact uncertainty caclulation class.
+
+    This is the class to perform uncertainty analysis on the outputs of a
+    climada.engine.impact.Impact() object.
+
+    Attributes
+    ----------
+    rp : list(int)
+        List of the chosen return periods.
+    calc_eai_exp : bool
+        Compute eai_exp or not
+    calc_at_event : bool
+        Compute eai_exp or not
+    metric_names : tuple(str)
+        Names of the impact output metris
+        ('aai_agg', 'freq_curve', 'at_event', 'eai_exp', 'tot_value')
+    value_unit : str
+        Unit of the exposures value
+    input_var_names : tuple(str)
+        Names of the required uncertainty input variables
+        ('exp_input_var', 'impf_input_var', 'haz_input_var')
+    exp_input_var : climada.engine.uncertainty.input_var.InputVar
+        Exposure uncertainty variable
+    impf_input_var : climada.engine.uncertainty.input_var.InputVar
+        Impact function set uncertainty variable
+    haz_input_var: climada.engine.uncertainty.input_var.InputVar
+        Hazard uncertainty variable
+    """
+
+    def __init__(self, exp_input_var, impf_input_var, haz_input_var):
+        """Initialize UncCalcImpact
+
+        Sets the uncertainty input variables, the impact metric_names, and the
+        units.
+
+        Parameters
+        ----------
+        exp_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.Exposure
+            Exposure uncertainty variable or Exposure
+        impf_input_var : climada.engine.uncertainty.input_var.InputVar or climada.entity.ImpactFuncSet
+            Impact function set uncertainty variable or Impact function set
+        haz_input_var : climada.engine.uncertainty.input_var.InputVar or climada.hazard.Hazard
+            Hazard uncertainty variable or Hazard
+
+        """
+
+        Calc.__init__(self)
+        self.input_var_names = ('exp_input_var', 'impf_input_var', 'haz_input_var')
+        self.exp_input_var =  InputVar.var_to_inputvar(exp_input_var)
+        self.impf_input_var =  InputVar.var_to_inputvar(impf_input_var)
+        self.haz_input_var =  InputVar.var_to_inputvar(haz_input_var)
+        self.metric_names = ('aai_agg', 'freq_curve', 'at_event',
+                             'eai_exp', 'tot_value')
+        self.value_unit = self.exp_input_var.evaluate().value_unit
+
+
+    def uncertainty(self,
+                    unc_sample,
```

Comment:
I'd prefer this method to return its own `UncOutput` extension, `UncImpactOutput`. Please see my comment at the bottom of unc_ouput.